packages/itwin/tree-widget/src/components/trees/stateless/common/UseHierarchyVisibility.ts

```diff
@@ -25,19 +26,19 @@ interface UseHierarchyVisibilityProps {
 
 interface UseHierarchyVisibilityResult {
   onCheckboxClicked: (node: PresentationHierarchyNode, checked: boolean) => void;
-  getCheckboxStatus: (node: PresentationHierarchyNode) => VisibilityStatus;
+  getCheckboxState: (node: PresentationHierarchyNode) => TreeNodeCheckboxState;
 }
 
 /** @internal */
 export function useHierarchyVisibility({ visibilityHandlerFactory }: UseHierarchyVisibilityProps): UseHierarchyVisibilityResult {
-  const statusMap = useRef(new Map<string, { node: PresentationHierarchyNode; status: VisibilityStatus; needsRefresh: boolean }>());
-  const [state, setState] = useState({
-    getCheckboxStatus: (_node: PresentationHierarchyNode): VisibilityStatus => ({ state: "hidden", isDisabled: true }),
+  const nodesStateMap = useRef(new Map<string, { node: PresentationHierarchyNode; state: TreeNodeCheckboxState; needsRefresh: boolean }>());
+  const [state, setState] = useState<UseHierarchyVisibilityResult>({
+    getCheckboxState: (_node: PresentationHierarchyNode): TreeNodeCheckboxState => ({ state: "off", isDisabled: true }),
     onCheckboxClicked: (_node: PresentationHierarchyNode, _checked: boolean): void => {},
   });
```

Comment:
Since you explicitly defined the type this should no longer be necessary.

```suggestion
  const [state, setState] = useState<UseHierarchyVisibilityResult>({
    getCheckboxState: () => ({ state: "off", isDisabled: true }),
    onCheckboxClicked: () => {},
  });
```